geomstats/datasets/sample_sdp_2d.py

```diff
@@ -0,0 +1,131 @@
+        Generate the dataset.
+
+        :return:
+        X: array-like, shape = [n_samples * n_classes, n_features]: data
+        Y: array-like, shape = [n_samples * n_classes, n_classes]: labels
+        """
+        X, Y = self.setup_data()
+        X, Y = self.data_helper.shuffle(X, Y)
+        return X, Y
+
+    def setup_data(self):
+        """
+        Generate the un-shuffled dataset.
+
+        :return:
+        X: array-like, shape = [n_samples * n_classes, n_features]: data
+        Y: array-like, shape = [n_samples * n_classes, n_classes]: labels
+        """
+        mean_covariance_eigenvalues = gs.random.uniform(
+            0.1, 5., (self.n_classes, self.n_features))
+        base_rotations = SpecialOrthogonal(n=self.n_features).random_gaussian(
+            gs.eye(self.n_features), 1, n_samples=self.n_classes)
+        # var_eigenvalues = gs.random.uniform(
+        #     .04, .06, (self.n_classes, self.n_features))
+        var_rotations = gs.random.uniform(
+            .5, .75, (self.n_classes))
+
+        # data
+        cov = gs.zeros(
+            (self.n_classes *
+             self.n_samples,
+             self.n_features,
+             self.n_features))
+        Y = gs.zeros((self.n_classes * self.n_samples, self.n_classes))
+        for i in range(self.n_classes):
+            cov[i * self.n_samples:(i + 1) * self.n_samples] = self.make_data(
+                base_rotations[i], gs.diag(
+                    mean_covariance_eigenvalues[i]), var_rotations[i])
+            Y[i * self.n_samples:(i + 1) * self.n_samples, i] = 1
+        return cov, Y
+
+    def make_data(self, eigenspace, eigenvalues, var):
+        """
+        Generate Gaussian data from mean matrix and variance.
+
+        :param eigenspace: array-like, shape = [n, n]
+        :param eigenvalues: array-like, shape = [n, n] (diagonal matrix)
+        :param var: float, variance of the wanted distribution.
+        :return: array-like, shape = [n, n]: data.
+        """
+        spd = SPDMatrices(n=self.n_features)
+        spd.set_eigensummary(eigenspace, eigenvalues)
+        spd_data = spd.random_gaussian(
+            var_rotations=var, n_samples=self.n_samples)
+        return spd_data
+
+    def make_data_noisy(self, eigenspace, eigenvalues, var, var_eigenvalues):
+        """
+        Generate noisy Gaussian data from mean matrix and variance.
+
+        :param eigenspace: array-like, shape = [n, n]
+        :param eigenvalues: array-like, shape = [n, n] (diagonal matrix)
+        :param var: float, variance of the wanted distribution.
+        :param var_eigenvalues: float, noise within the distribution.
+        :return: array-like, shape = [n, n]: data.
+        """
+        spd = SPDMatrices(n=self.n_features)
+        spd.set_eigensummary(eigenspace, eigenvalues)
+        spd_data = spd.random_gaussian_noisy(
+            var_rotations=var, noise=var_eigenvalues, n_samples=self.n_samples)
+        return spd_data
+
+
+class DataHelper():
```

Comment:
No ()

Reply:
I'd rather leave them wrapped together, a design choice which could facilitate maintaining utility functions common to different settings. Perhaps a standalone file in the `datasets` folder could hold this single helper class, which could then be expanded upon.

Reply:
Or: a standalone file `datasets/utils.py` that holds these functions without the need of the class? The functions would be called with `datasets.utils.shuffle()` syntax, which wraps them together already.

But I let you decide.